src/internal/snapshot/recovery.go

```diff
@@ -475,3 +475,31 @@ func (s *Snapshotter) RestoreSnapshot(rctx context.Context, id SnapshotID, opts
 
 	return nil
 }
+
+// DropSnapshot deletes a snapshot and everything it references, allowing any data it closed over to
+// be garbage collected.
+func (s *Snapshotter) DropSnapshot(ctx context.Context, id SnapshotID) error {
+	if err := dbutil.WithTx(ctx, s.DB, func(ctx context.Context, tx *pachsql.Tx) error {
+		row, err := snapshotdb.GetSnapshot(ctx, tx, int64(id))
+		if err != nil {
```

Comment:
this can be on the same line, just as the following if clauses

Reply:
We need to read `row` below.

Reply:
My bad. I missed that